### PR TITLE
Publicize mocks of our interfaces

### PR DESCRIPTION
**Pull request by muvaf**
### Description of your changes

They are helpful for the tests we want to write for the following checks where the test object should adhere to `Managed` interface but not the actual type:
```
cr, ok := mg.(*v1alpha2.RDSInstance)
if !ok {
	return resource.ExternalObservation{}, errors.New(errNotRDSInstance)
}
```

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [x] Updated any relevant [documentation], [examples], or [release notes].
- [x] Updated the RBAC permissions in [`clusterrole.yaml`] to include any new types.

[documentation]: https://github.com/crossplaneio/crossplane/tree/master/docs
[examples]: https://github.com/crossplaneio/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplaneio/crossplane/tree/master/PendingReleaseNotes.md
[`clusterrole.yaml`]: https://github.com/crossplaneio/crossplane/blob/master/cluster/charts/crossplane/templates/clusterrole.yaml

### CODE REVIEW

**Comment by negz** [comment]
This seems like a good direction to me. Does it block your work on 0.4 @muvaf? It might be nice to move it into a distinct `fake` package, so we have `fake.Managed{}` or similar.

**Comment by negz** [comment]
I don't think this is a priority, but I'd still be happy to approve this PR if it was rebased and the interfaces were moved into a `fake` package. Let me know if you're still interested in moving forward with this @muvaf.

**Comment by muvaf** [comment]
@negz I just ran into another case where I need these guys in the tests. Updated the PR with your suggestion, it should be ready to merge.

**Comment by negz** [comment]
@muvaf Looks like there's a merge conflict.

File: pkg/resource/api_test.go [hunk 1]
```diff
@@ -20,6 +20,8 @@ import (
 	"context"
 	"testing"
 
+	"github.com/crossplaneio/crossplane-runtime/pkg/resource/fake"
```
Comment (negz) [inline pkg/resource/api_test.go]: Would prefer if this import wasn't floating by itself.

File: pkg/resource/api_test.go [hunk 2]
```diff
@@ -71,13 +73,13 @@ func TestCreate(t *testing.T) {
 				client: &test.MockClient{
 					MockCreate: test.NewMockCreateFn(errBoom),
 				},
-				typer: MockSchemeWith(&MockClaim{}, &MockClass{}, &MockManaged{}),
+				typer: fake.MockSchemeWith(&fake.MockClaim{}, &fake.MockClass{}, &fake.MockManaged{}),
```
Comment (negz) [inline pkg/resource/api_test.go]: `fake.MockFoo` feels a bit stutter-ey.  Now that these are in their own package I'd prefer they were named just `fake.Scheme`, `fake.Claim`, etc.

File: pkg/resource/api_test.go (hunk 1 above)
Comment (negz) [reply pkg/resource/api_test.go]: Looks like a few other imports have this issue too.

**Comment by negz** [comment]
LGTM in general, but I'd prefer if we could remove the `Fake` prefix from the type names (since they're now in package `fake`). Could you also add a package doc comment to the new `fake` package? Take a look at the other package doc strings for examples.

**Comment by muvaf** [comment]
@negz Thanks for the naming suggestion! It should be ready to merge.

**Pull request merged**